test/integration/fake_upstream.cc

```diff
@@ -447,7 +447,7 @@ FakeUpstream::FakeUpstream(Network::TransportSocketFactoryPtr&& transport_socket
       api_(Api::createApiForTest(stats_store_)), time_system_(time_system),
       dispatcher_(api_->allocateDispatcher("fake_upstream")),
       handler_(new Server::ConnectionHandlerImpl(*dispatcher_)),
-      allow_unexpected_disconnects_(false), read_disable_on_new_connection_(true),
+      allow_unexpected_disconnects_(true), read_disable_on_new_connection_(true),
```

Comment:
really my only question is if we should remove allow_unexpected_disconnects_ and set_allow_unexpected_disconnects() entirely, which will likely break downstream test builds, but it's only worth keeping around if someone really wants some test to allow_unexpected_disconnects_(false) explicitly.  Maybe a follow-up so we can roll only that part back if someone wants that?

Reply:
I think doing it in a follow up makes sense. Do we not have any tests that want to test for unexpected disconnects right now?

Reply:
not that I know of?  I mean we could arguably set it false for every test which didn't previously have it set true, but I think most tests which wanted to know they got a complete request waited for a complete requests, ditto response.

Reply:
I don't think that's necessary, probably fine to just get rid of for now unless someone complains

Reply:
I'm fine with a follow up but I would be fine just getting rid of it all now.